docs/examples/multiple-agent-shared-memory.md

```diff
@@ -0,0 +1,93 @@
+This example shows how to use one `Agent` to load content into `TaskMemory` and get that content from another `Agent` using `VectorStoreClient`.
+
+The first `Agent` uses a remote vector store (`MongoDbAtlasVectorStoreDriver` in this example) to handle memory operations. The second `Agent` uses the `VectorStoreClient` with the same `MongoDbAtlasVectorStoreDriver` to get the data.
+
+The `MongoDbAtlasVectorStoreDriver` assumes that you have a vector index configured where the path to the content is called `vector`, and the number of dimensions set on the index is `1536`.
+
+```python
+from dotenv import load_dotenv
+from griptape.tools import WebScraper, VectorStoreClient, TaskMemoryClient
+from griptape.structures import Agent
+from griptape.drivers import AzureOpenAiChatPromptDriver, AzureOpenAiEmbeddingDriver, MongoDbAtlasVectorStoreDriver
+from griptape.engines import VectorQueryEngine, PromptSummaryEngine, CsvExtractionEngine, JsonExtractionEngine
+from griptape.memory import TaskMemory 
+from griptape.artifacts import TextArtifact
+from griptape.memory.task.storage import TextArtifactStorage
+
+import os
+
+load_dotenv()
+
+AZURE_OPENAI_ENDPOINT = os.getenv("AZURE_OPENAI_ENDPOINT")
+
+MONGO_CONNECTION_STRING = os.getenv('MONGO_CONNECTION_STRING')
+MONGO_DATABASE_NAME = os.getenv('MONGO_DATABASE_NAME')
+MONGO_COLLECTION_NAME = os.getenv('MONGO_COLLECTION_NAME')
+
+
+azure_embedding_driver = AzureOpenAiEmbeddingDriver(
+    model='text-embedding-ada-002',
+    azure_endpoint=AZURE_OPENAI_ENDPOINT,
+    azure_deployment='text-embedding-ada-002'
+)
+
+azure_prompt_driver = AzureOpenAiChatPromptDriver(
+    model='gpt-4',
+    azure_endpoint=AZURE_OPENAI_ENDPOINT,
+    azure_deployment='gpt-4'
+)
+
+mongo_driver = MongoDbAtlasVectorStoreDriver(
+    connection_string=MONGO_CONNECTION_STRING,
+    database_name=MONGO_DATABASE_NAME,
+    collection_name=MONGO_COLLECTION_NAME,
+    embedding_driver=azure_embedding_driver,
+)
+
+vector_engine = VectorQueryEngine(
+    vector_store_driver=mongo_driver,
+    prompt_driver=azure_prompt_driver,
+)
+vector_store_tool = VectorStoreClient(
+    description='Can be used for all user queries',
+    query_engine=vector_engine,
+    off_prompt=False
+)
+
+task_memory = TaskMemory(
+    artifact_storages={
+        TextArtifact: TextArtifactStorage(
+            query_engine=vector_engine,
+            summary_engine=PromptSummaryEngine(
+                prompt_driver=azure_prompt_driver
+            ),
+            csv_extraction_engine=CsvExtractionEngine(
+                prompt_driver=azure_prompt_driver
+            ),
+            json_extraction_engine=JsonExtractionEngine(
+                prompt_driver=azure_prompt_driver
+            )
+        ),
+    },
+)
+
+loader = Agent(
+    tools=[
+        WebScraper()
+    ],
+    prompt_driver=azure_prompt_driver,
+    embedding_driver=azure_embedding_driver,
+    task_memory=task_memory,
+)
+asker = Agent(
+    tools=[
+        vector_store_tool
```

Comment:
moving this from a private thread:
if i want both agents to use `TaskMemory` and the asker to use `TaskMemoryClient`, then they need to share `MetaMemory` between them AND the same instance of `TaskMemory` because of the `dict` of `namespace_storage` that uses the `artifact_namespace` as the key. the `VectorStoreClient` is able to pull the information assuming there is no namespace applied.

Reply:
Ah right, understood. Thanks for the explanation.

Reply:
the above should probably be discussed more. I can see it being useful for two Structures to essentially have the same memory and access to the same data via `TaskMemory/Client`

Reply:
Should we use that example then? To show how you can share Task and Meta Memory? Obviously there are limitations with their current in-memory implementations, but I think it demonstrates an important concept.

Reply:
updated the example